CMakeLists.txt

```diff
@@ -115,11 +115,15 @@ if(ENABLE_LTO)
 endif()
 
 if(ENABLE_OPENMP)
-    find_package( OpenMP REQUIRED )
-    include(FindOpenMP)
-    if(OPENMP_FOUND)
+    if(APPLE)
+        set(OpenMP_CXX_INCLUDE_DIR "/usr/local/opt/libomp/include")
+        set(OpenMP_C_INCLUDE_DIR "/usr/local/opt/libomp/include")
+    endif()
+    find_package(OpenMP REQUIRED)
+    if(OpenMP_FOUND)
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
-        message(STATUS "found OpenMP, compiling with flags: " ${OpenMP_CXX_FLAGS} )
+        message(STATUS "found OpenMP, compiling with flags: " ${OpenMP_CXX_FLAGS})
+        include_directories(${OpenMP_CXX_INCLUDE_DIRS})
```

Comment:
Let me ask one more (potentially stupid) question. Does it work without this change on macOS after the change on line 119-120?

Reply:
The `include_directories` is necessary to be able to use the `omp_get_max_threads()` function.
Since we don't directory call any functions from OpenMP, we didn't need the include path to be added, but for this change it is needed. So yes, it works without this change as well, it is only needed to display the number of threads.

However, I was just looking at the master branch again and debugging the OpenMP variables, and found it actually all to be correct, so that would mean that the `OpenMP_CXX_INCLUDE_DIR` is unnecessary.

I wonder how it had trouble finding the omp.h file though, I might not have cleared the CMake cache before a build or bricked OpenMP by recompiling it for am64.

In any case, I think we can do without the whole
```
set(OpenMP_CXX_INCLUDE_DIR "/usr/local/opt/libomp/include")
set(OpenMP_C_INCLUDE_DIR "/usr/local/opt/libomp/include")
```

Stuff, and just keep
```
include_directories(${OpenMP_CXX_INCLUDE_DIRS})
```
Around in the CMakeLists.txt

Reply:
Ahaaa... So I just reverted that change and get png compilation errors because (I THINK) it is trying to use the one from homebrew...
I guess that is what this whole thing was about, and it brings back some of my sanity hehe.

Anyways, I will look into this another time, I am off for a while during Christmas.

Happy holidays everyone ;)

Reply:
Sounds good.

Reply:
I moved the `OpenMP_CXX_INCLUDE_DIRS` to the end of the `include_directories` call in `src/CMakeLists.txt` and added a comment that it should be last for macOS.

The joy of cross platform development I guess ;)

Reply:
And looking at it now, it might not even need to be last, the include_directories probably just shouldn't be before adding the vendored libraries stuff...
I guess I really need a break from programming for a while ;)